src/modal/modal.js

```diff
@@ -555,20 +557,65 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
 
         openedWindows.top().value.modalDomEl = angularDomEl;
         openedWindows.top().value.modalOpener = modalOpener;
+
+        applyAriaHidden(angularDomEl[0]);
+
+        function applyAriaHidden(el) {
+          if (!el || el.tagName === 'BODY') {
+            return;
+          }
+
+          getSiblings(el).forEach(function(sibling) {
+            var elemIsAlreadyHidden = sibling.getAttribute('aria-hidden') === 'true',
+              ariaHiddenCount = 
+                parseInt(sibling.getAttribute(ARIA_HIDDEN_ATTRIBUTE_NAME), 10) || 
+                  elemIsAlreadyHidden ? 1 : 0;
```

Comment:
This is very confusing - mixing the logical operator with the ternary makes this difficult to parse. What is the intention here? It looks like it is to use the attribute value or if it does not have a value and the element is already hidden, then to use 1 or 0. This can be written better if that is the case.